sql/core/src/main/scala/org/apache/spark/sql/execution/Queryable.scala

```diff
@@ -34,4 +36,23 @@ private[sql] trait Queryable {
         s"Invalid tree; ${e.getMessage}:\n$queryExecution"
     }
   }
+
+  /**
+   * Prints the plans (logical and physical) to the console for debugging purposes.
+   * @since 1.3.0
+   */
+  def explain(extended: Boolean): Unit = {
+    val explain = ExplainCommand(queryExecution.logical, extended = extended)
+    DataFrame(sqlContext, explain).queryExecution.executedPlan.executeCollect().foreach {
```

Comment:
maybe we can simplify this to `sqlContext.executePlan(explain).excutedPlan......`